core/chains/evm/logpoller/log_poller.go

```diff
@@ -277,7 +284,8 @@ func (lp *logPoller) getReplayFromBlock(ctx context.Context, requested int64) (i
 
 func (lp *logPoller) run() {
 	defer close(lp.done)
-	tick := time.After(0)
+	logPollTick := time.After(0)
+	blockPruneTick := time.After(0)
```

Comment:
Just curiosity, but can you explain this? AFAIK go's `select` nondeterministically selects a branch (see https://go.dev/ref/spec#Select_statements):

> If one or more of the communications can proceed, a single one that can proceed is chosen via a uniform pseudo-random selection.

What's the benefit of doing this vs. setting them to the appropriate durations straight away here?

Reply:
it is indeed random - no real benefit imo. I didn't see much of a difference between pruning or begin polling on lp boot